slides/aml-25-042518-time-series/index.html

```diff
@@ -0,0 +1,952 @@
+
+
+???
+
+---
+
+
+.left-column[
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; Evenly spaced
+![:scale 95%](images/evenly spaced data.png)
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;CO2 Measurements
+- Today only evenly spaced
+
+]
+.right-column[
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Unevenly spaced
+![:scale 100%](images/unevenly spaced data.png)
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Volcanic Eruption Deaths
+- For uneven: sometimes aggregates over fixed periods make sense
+]
+
+???
+
+---
+class: center, middle
+
+# Tasks 
+
+???
+
+---
+
+# 1d Forecasting
+
+.center[![:scale 80%](images/1d forecasting.png)]
+
+???
+
+---
+
+# Nd Forecasting
+
+.center[![:scale 80%](images/nd forecasting.png)]
+
+???
+
+---
+class: spacious
+
+# Feature-based forecasting &nbsp; (expanatory variables)
+
+.center[![:scale 80%](images/feature based forecasting.png)]
+
+???
+
+---
+
+# Time Series Classification
+
+.center[![:scale 70%](images/time series classification.png)]
+
+???
+
+---
+
+
+# Tasks
+
+- 1d time-series forecasting
+  - Number of customers (naive)
+- Multivariate time-series -- forecast all
+  - Stock market
+- Multivariate time-series -- forecast one
+  - Number of customers (with features)
+- Time-series classification (1d or Nd)
+  - Activity from smart phone sensors
+
+???
+
+---
+class: center, middle
+
+# Data reading / munging with Pandas
+
+???
+
+---
+
+# Parse Dates
+
+.smaller[
+```python
+url = "ftp://aftp.cmdl.noaa.gov/products/trends/co2/co2_weekly_mlo.txt"
+names = ["year", "month", "day", "year_decimal", "co2", "days", "1 yr ago", 
+          "10 yr ago", "since 1800"]
+maunaloa = pd.read_csv(url, skiprows=49, header=None, delim_whitespace=True,
+                       names=names, na_values=[-999.99])
+maunaloa.head()
+```
+]
+
+.center[![:scale 80%](images/parse dates.png)]
+
+???
+
+---
+
+# Parse Dates
+
+.smaller[
+```python
+maunaloa = pd.read_csv(url, skiprows=49, header=None, delim_whitespace=True,
+                       names=names, parse_dates=[[0, 1, 2]], na_values=[-999.99])
+maunaloa.head()                       
+```
+]
+
+.center[![:scale 80%](images/parse dates 2.png)]
+
+???
+
+---
+
+# Time Series Index
+
+.smaller[
+```python
+maunaloa = pd.read_csv(url, skiprows=49, header=None, delim_whitespace=True,
+                       names=names, parse_dates=[[0, 1, 2]], na_values=[-999.99],
+                       index_col="year_month_day")
+maunaloa.head()                       
+```
+]
+
+.center[![:scale 80%](images/time series index.png)]
+
+
+???
+
+---
+class: spacious
+
+.left-column[
+
+.smaller[
+```python
+maunaloa.co2.head()
+```
+```python
+year_month_day
+1974-05-19    333.34
+1974-05-26    332.95
+1974-06-02    332.32
+1974-06-09    332.18
+1974-06-16    332.37
+Name: co2, dtype: float64
+```
+```python
+maunaloa.head()                       
```

Comment:
should be maunaloa.co2.plot but you can also just remove it.

Reply:
done